files/src/main/scala/io/iteratee/files/SuspendableFileModule.scala

```diff
@@ -0,0 +1,100 @@
+package io.iteratee.files
+
+import cats.MonadError
+import io.iteratee.{ Enumerator, Module }
+import io.iteratee.internal.Step
+import java.io.{
+  BufferedInputStream,
+  BufferedReader,
+  Closeable,
+  File,
+  FileInputStream,
+  FileReader,
+  InputStream,
+  InputStreamReader
+}
+import java.util.zip.{ ZipEntry, ZipFile }
+import scala.Predef.genericArrayOps
+import scala.collection.JavaConverters._
+
+/**
+ * File operations for contexts that are suspendable and for which recursive
+ * monadic binding is stack safe.
+ */
+trait SuspendableFileModule[F[_]] extends FileModule[F] {
+    this: Module[F] { type M[f[_]] <: MonadError[f, Throwable] } =>
+  protected def captureEffect[A](a: => A): F[A]
+  private[this] def close(c: Closeable): F[Unit] = captureEffect(c.close())
+
+  final def readLines(file: File): Enumerator[F, String] =
+    Enumerator.liftM(captureEffect(new BufferedReader(new FileReader(file))))(F).flatMap { reader =>
+      new LineEnumerator(reader).ensure(close(reader))(F)
+    }(F)
+
+  final def readLinesFromStream(stream: InputStream): Enumerator[F, String] =
+    Enumerator.liftM(captureEffect(new BufferedReader(new InputStreamReader(stream))))(F).flatMap { reader =>
+      new LineEnumerator(reader).ensure(close(reader))(F)
+    }(F)
+
+  final def readBytes(file: File): Enumerator[F, Array[Byte]] =
+    Enumerator.liftM(captureEffect(new BufferedInputStream(new FileInputStream(file))))(F).flatMap { stream =>
+      new ByteEnumerator(stream).ensure(close(stream))(F)
+    }(F)
+
+  final def readBytesFromStream(stream: InputStream): Enumerator[F, Array[Byte]] =
+    Enumerator.liftM(captureEffect(new BufferedInputStream(stream)))(F).flatMap { stream =>
+      new ByteEnumerator(stream).ensure(close(stream))(F)
+    }(F)
+
+  final def readZipStreams(file: File): Enumerator[F, (ZipEntry, InputStream)] =
```

Comment:
shame we can't share these.

Like a common `protected def catch[A](a: => A): F[A]` and in Suspendable `catch == capture` in `NonSuspendable` `catch == catchNonFatal`